added sample for tkinter live-view&capture and retrieving supported framesizes/intervals

emdete opened the pull request:
this branch mainly contains two features:
- capture program with tk gui (mainly developed to be used with a cheap slide scanner where you want to change resolution while live-view and capture)
- add getters for frame sizes and intervals

if needed i can reduce the commits and separate the two features into two branches/pull requests.

regards, michael

Commit:
use canvas

Commit:
make work

Commit:
improve scanning, add keys

Commit:
add comments, rename

Commit:
check existense always

Commit:
add pause image

Commit:
rename&reorder

Commit:
add controls

Commit:
add cam capture with gui

Commit:
fix change video device menu

Commit:
add persistance config

Commit:
adjust comments & doc

Commit:
fix startup

Commit:
adapt README.md

Commit:
add 'x' to quit programm

Commit:
fix typos in README.md

Commit:
add image of scanner

jnohlgard commented:
You need to get rid of the indentation changes, it is very difficult to review when every line of the file has been changed.

Commit:
fix dev name conf

Commit:
prepare rotate

Commit:
rename members to be easier to understand

Commit:
add more keys

Commit:
adjust comment

Commit:
use new names

Commit:
run indent on c code

Commit:
add getter for frame size & interval

Commit:
add new feature to sample code

Commit:
collect v4l library switch to one location

emdete commented:
Hi,

you are right, i changed the code formatting alot as i saw it was not consitent (and as a python developer i agree that whitespace matters). so i edit history to have this commits:

```
run indent on c code
add getter for frame size & interval
add new feature to sample code
collect v4l library switch to one location
```

the whole reformatting is now a single step done by `indent -br -brf -brs -nce -nut -ppi 2 v4l2capture.c` and my changes now are quite clear.
i also renamed the README as it is markdown and with that extension is properly displayed in github. i removed files that are not needed anymore. i hope that is fine with you.

regards,
michael

Commit:
remove unused files

jnohlgard commented:
You need to make a separate pull request for the indentation changes, if you look at the "Files Changed" tab of this PR, you will see that it is quite difficult to review this when everything has changed. Don't mix scripted automatic indentation and whitespace changes to the whole tree with functional changes.

Either:
- Create a separate branch, do all the whitespace changes you like, open a new PR, then rebase this change on top of that indentation change branch.

Or:
- Make all the functional changes to the current `master`, without touching the indentation of unrelated functions. Then make another branch based on that PR, run indent and make a new PR with the indentation changes.
